src/lib/Board.ts

```diff
@@ -0,0 +1,111 @@
+import Instance from './Instance';
+import Coordinate from './Coordinate';
+import Cell from './Cell';
+
+const BOARD_SIZE = 9;
+
+const clone = <T>(array: T[]): T[] => [...array];
+const removeItem = <T>(array: T[], idx: number): T[] => [
+  ...clone(array).splice(0, idx),
+  ...clone(array).splice(idx + 1),
+];
+const addItem = <T>(array: T[], idx: number, item: T): T[] => [
+  ...clone(array).splice(0, idx),
+  item,
+  ...clone(array).splice(idx),
+];
+
+export default class SudokuBoard extends Instance {
+  private _board: Cell[];
+
+  constructor() {
+    super();
+    this._board = [];
+
+    // Initialize the board with 0s
+    for (let i = 0; i < BOARD_SIZE; i++) {
+      for (let j = 0; j < BOARD_SIZE; j++) {
+        this._board.push(new Cell(new Coordinate(i, j)));
+      }
+    }
+  }
+
+  fetchCellsInSquare(row: number, col: number) {
+    const squareRow = Math.floor(row / 3);
+    const squareCol = Math.floor(col / 3);
+
+    const cells = [];
+    for (let i = 0; i < 3; i++) {
+      for (let j = 0; j < 3; j++) {
+        cells.push(this.fetchCell(squareRow * 3 + i, squareCol * 3 + j).cell);
+      }
+    }
+    return cells;
+  }
+
+  fetchCell(row: number, col: number) {
+    const cell = this._board.find((cell) =>
+      cell.coordinate.equals(new Coordinate(row, col))
+    );
```

Comment:
```suggestion
    const cellCoordinate = new Coordinate(row, col);
    const cell = this._board.find((cell) =>
      cell.coordinate.equals(cellCoordinate)
    );
```
Avoid construction of a new coordinate per cell, in favour of a constant instance